Use STATE_CLASS_TOTAL_INCREASING value by default and add additional entities

JacoboSanchez opened the pull request:
I am making some tests on your new async implementation and made some modifications for myself so I am making this PR with this changes just in case you find any of them useful. 

First of all I have added STATE_CLASS_TOTAL_INCREASING which was added to home assistant in order to manage this kind incremental values to avoid using a fake last_reset attribute with value 0. As daily yield is the only one (if I do not miss anything) that is not working this way I just use the total increasing as default and modified daily yield as an exception.

Also I added two more entities I found missing to use the home assistant energy system (grid export and consumption), as all other (production and battery charge/discharge) are already available

For my case of use I am planning to create a few more entity sensors in my local version like state of capacity, power meter active power and input power to avoid extracting them with templates but they are not included in this PR

Emilv2 commented:
Looks good to me. I think in the end all of the attributes (or at least most of them) should be entities, which then can be enabled in the UI. But I haven't looked in to how to do that.

JacoboSanchez commented:
I think there must be a balance. It sounds good to have everything available and configurable but at the end that adds complexity and requires effort. Rarely used things may be left as attributes and most used ones can be added by default and that may give a decent user experience for most people. The problem there is to decide which ones are the most used :)

Attributes are great to store additional info... the problem is that they are not really accepted by most of the cards and they are hard to understand for new HA users as few examples in the documentation/forums use them.

JacoboSanchez commented:
I have made [additional changes](https://github.com/JacoboSanchez/huawei_solar/commit/a856a50348f7bf915a211643d6649e0480004fb8#diff-a3af1044ed62c1eda8f7526cb8b7655afcc4c1b8e8fa47145388a92189b81cf0) in my personal repo to add the entities I would like to have and changed some default icons. However I removed them from the attributes list and that would break other user's configurations so I am not making a PR for those modifications. 
A question about that last comment... If I keep the attributes while adding the entities would make the integration to perform two modbus calls for those values?
